pkg/trait/builder.go

```diff
@@ -32,6 +34,8 @@ type builderTrait struct {
 	BaseTrait `property:",squash"`
 	// Enable verbose logging on build components that support it (e.g. Kaniko build pod).
 	Verbose *bool `property:"verbose" json:"verbose,omitempty"`
+	// A list of properties to be provided to the build task
+	BuildTimeProperties []string `property:"build-time-properties" json:"buildTimeProperties,omitempty"`
```

Comment:
Can I suggest:

    Properties []string `property:"properties" json:"Properties,omitempty"`

It's quite clear in the scope of the builder trait that these properties are applied at build time.